chainercv/utils/iterator/apply_semantic_segmentation_link.py

```diff
@@ -0,0 +1,63 @@
+import numpy as np
+
+from chainercv.utils.iterator.split_iterator import split_iterator
+
+
+def apply_semantic_segmentation_link(target, iterator, hook=None):
+    """Apply a semantic segmentation link to an iterator
+
+    This function applies a semantic segmentation link to an iterator.
+    It stacks the outputs of the semantic segmentation link
+    into :obj:`pred_labels`.
+    This function also stacks the values returned by the iterator.
+    These values can be used for evaluation.
+
+    Args:
+        target (chainer.Link): A semantic segmentation link. This link must
+            have :meth:`predict` method which take a list of images and returns
+            :obj:`labels`.
+        iterator (chainer.Iterator): An iterator. Each sample should have
+            an image as its first element. This image is passed to
+            :obj:`target`. The rests are stacked into :obj:`gt_values`.
+        hook: A callable which is called after each iteration.
+            :obj:`pred_labels` and :obj:`gt_values` are passed as arguments.
+            Note that these values do not contain data from the previous
+            iterations.
+
+    Returns:
+        An iterator and a tuple:
+        This function returns :obj:`pred_labels` and :obj:`gt_values`.
+        :obj:`gt_values` is a tuple of iterators. Each iterator corresponds
+        to an value of samples from the iterator.
+        For example, if the iterator returns batches of :obj:`img, val0, val1`,
```

Comment:
a batch of :obj:`img, val0, val1` instead of "batches"?

Note: next(it) == [(img, val0, val1)]